dask_match/core.py

```diff
@@ -290,38 +345,114 @@ def _meta(self):
     def _kwargs(self):
         return {}
 
+    def _broadcast_dep(self, dep: Expr):
+        # Checks if a dependency should be broadcasted to
+        # all partitions of this `Blockwise` operation
+        return (
+            not isinstance(dep, BlockwiseArg)
+            and dep.npartitions == 1
+            and dep.ndim < self.ndim
+        )
```

Comment:
I'm curious, why this relationship rather than `dep.is_broadcastable`?

What would happen in the case where self.ndim is also equal to one?  Would it even matter?

Reply:
My intuition is that making an `is_broadcastable` property will end up being smoother.  I don't have great evidence for that though, just intuition.

Reply:
Oh, this is the DataFrame + Series thing?

Reply:
This could be wrong? I assumed that you cannot “broadcast” a series over a series or a dataframe over a dataframe. If both sides have the same dimension you would need to align divisions rather than broadcast.

Reply:
Yeah, ok, nm, that makes sense.